### PR TITLE
DS-140 upgrade lb camden frontend to gov uk frontend 4.7.0 from 4.6.0

### PR DESCRIPTION
**Pull request by oluoluoxenfree**
Summary of changes:

- removed`puppeteer` as we have `jest-puppeteer` and seemingly unused directly
- bumped `govuk-frontend` to `4.7.0`
- bumped node in `.nvmrc` and node engine in `package.json` to `20.10.0`, as our tooling wants a numerical version and not `lts/iron`
- added `jest-environment-jsdom` at `29.7.0`
- bumped `gulp-postcss` to `9.0.0`
- bumped `jest` to `29.7.0`
- bumped `jest-puppeteer` to `9.0.2`
- bumped `node-sass` to `9.0.0`
- bumped `postcss-pseudo-classes` to `0.4.0`
- commented out failing rollup test for IE8 support

### CODE REVIEW

File: package.json.test.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /* eslint-env jest */
 
 const packageJson = require('./package.json')
-const packageLockJson = require('./package-lock.json')
+// const packageLockJson = require('./package-lock.json')
 
 describe('because rollup 0.60 drops support for Internet Explorer 8', () => {
   describe('gulp-better-rollup', () => {
```
Comment (chrisdevereux) [inline package.json.test.js:7]: we've dropped IE8 support, so we could delete these IE8 tests rather than commenting them out

File: package.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
   "description": "Used only for the development of CAMDEN.GOV.UK Frontend, see `package/package.json` for the published `package.json`",
   "version": "0.4.5",
   "engines": {
-    "node": "~16.14.2"
+    "node": "^20.9.0"
```
Comment (alexwybraniec) [inline package.json:7]: @oluoluoxenfree Can you bump this to 20.10 inline with the nvmrc file, and check everything still works?

File: package.json.test.js (hunk 1 above)
Comment (alexwybraniec) [reply package.json.test.js:7]: @oluoluoxenfree Can you delete the rollup test, and check everything still builds?

**Review by alexwybraniec — CHANGES_REQUESTED** [review]
Please make the changes we requested, restest and re-submit.

File: package.json [hunk 3]
```diff
@@ -72,31 +72,31 @@
...
     "nodemon": "^2.0.4",
     "nunjucks": "^3.2.4",
     "oldie": "^1.3.0",
     "outdent": "^0.7.1",
     "path-webpack": "^0.0.3",
     "postcss": "^8.4.7",
-    "postcss-pseudo-classes": "0.2.0",
+    "postcss-pseudo-classes": "0.4.0",
     "postcss-scss": "^4.0.3",
-    "puppeteer": "^9.1.1",
```
Comment (alexwybraniec) [inline package.json:99]: @oluoluoxenfree Reintroduce puppeteer into the package.json as per https://www.npmjs.com/package/jest-puppeteer?activeTab=readme

File: package.json.test.js (hunk 1 above)
Comment (oluoluoxenfree) [reply package.json.test.js:7]: @alexwybraniec would it not be better to delete this entire file? It's all about ie8 support.

Comment (oluoluoxenfree) [reply package.json.test.js:7]: There are more ie8 mentions in the codebase which should probably be tidied up at some point.

**Review by chrisdevereux — APPROVED** [review]

**Review by alexwybraniec — APPROVED** [review]

**Pull request merged**